var/spack/repos/builtin/packages/camp/package.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Camp(CMakePackage, CudaPackage):
+    """
+    Compiler agnostic metaprogramming library providing concepts,
+    type operations and tuples for C++ and cuda
+    """
+
+    homepage = "https://github.com/LLNL/camp"
+    git      = "https://github.com/LLNL/camp.git"
+
+    version('master', branch='master', submodules='True')
+    version('0.1.0', tag='v0.1.0', submodules='True')
+
+    depends_on('cmake@3.8:', type='build')
+    depends_on('cmake@3.9:', type='build', when="+cuda")
```

Comment:
(no action required) as an aside: these two constraints 

```
depends_on('cmake@3.8:', type='build')
depends_on('cmake@3.9:', type='build', when="+cuda")
```

Can trip up Spack's current greedy concretization algorithm; namely the combination of a universal constraint and a constraint with a `when` condition. This is very unlikely to actually result in a problem because Spack always chooses the latest possible version of a package. It would be more likely to cause a problem if these constraints set upper bounds. More on this at https://github.com/spack/spack/issues/5465#issuecomment-332027738 (under item [1] of https://github.com/spack/spack/wiki/Issue-Disambiguation).

Reply:
That is good to know. This code was actually stolen from the chai package file, and I kept it here for consistency since long term I want build care with both chai and camp as dependencies